nautobot/dcim/models/devices.py

```diff
@@ -1375,3 +1386,353 @@ def clean(self):
             raise ValidationError(
                 {"controller": "Controller device group must have the same controller as the parent group."}
             )
+
+
+#
+# Modules
+#
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class ModuleType(PrimaryModel):
+    """
+    A ModuleType represents a particular make (Manufacturer) and model of Module. A Module can represent
+    a line card, supervisor, or other interchangeable hardware component within a ModuleBay.
+
+    ModuleType implements a subset of the features of DeviceType.
+
+    Each ModuleType can have an arbitrary number of component templates assigned to it,
+    which define console, power, and interface objects. For example, a Cisco WS-SUP720-3B
+    ModuleType would have:
+
+      * 1 ConsolePortTemplate
+      * 2 InterfaceTemplates
+
+    When a new Module of this type is created, the appropriate console, power, and interface
+    objects (as defined by the ModuleType) are automatically created as well.
+    """
+
+    manufacturer = models.ForeignKey(to="dcim.Manufacturer", on_delete=models.PROTECT, related_name="module_types")
+    model = models.CharField(max_length=CHARFIELD_MAX_LENGTH)
+    part_number = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH, blank=True, help_text="Discrete part number (optional)"
```

Comment:
Does `discrete` mean standalone here?

Reply:
In this context, it's more that it's a unique/distinct part number (IE: the manufacturer SKU)